thunder/clang/langctx.py

```diff
@@ -30,19 +30,19 @@ def get_method(self, id: str, *args, **kwargs) -> Callable:
         #       not exist.
         inps, _ = tree_flatten((args, kwargs))
 
-        has_tensor_input: bool = False
+        has_proxy_input: bool = False
         for x in inps:
-            if isinstance(x, TensorProxy):
-                has_tensor_input = True
+            if isinstance(x, TensorProxy) or isinstance(x, NumberProxy):
+                has_proxy_input = True
                 break
 
-        if has_tensor_input:
+        if has_proxy_input:
```

Comment:
This is where I'm using `clang` to explicitly handle dtype promotion for number proxies.